### PR TITLE
[pipeline](rpc) support closure reuse in pipeline exec engine

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Support closure reuse like no pipeline exec engine reduce new operator in exchange sink buffer

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

File: be/src/vec/sink/vdata_stream_sender.h [hunk 1]
```diff
@@ -525,11 +492,23 @@ class PipChannel final : public Channel {
         _buffer->register_sink(_fragment_instance_id);
     }
 
+    pipeline::SelfDeleteClosure<PTransmitDataResult>* get_closure(
+            InstanceLoId id, bool eos, vectorized::BroadcastPBlockHolder* data) {
+        if (!_closure) {
```
Comment (Gabriel39) [inline be/src/vec/sink/vdata_stream_sender.h:497]: Need to use a lock

File: be/src/vec/sink/vdata_stream_sender.h [hunk 2]
```diff
@@ -525,11 +492,23 @@ class PipChannel final : public Channel {
...
+        _closure->init(id, eos, data);
+        return _closure;
+    }
+
 private:
     friend class VDataStreamSender;
 
     pipeline::ExchangeSinkBuffer* _buffer = nullptr;
     bool _eos_send = false;
+    pipeline::SelfDeleteClosure<PTransmitDataResult>* _closure = nullptr;
```
Comment (Gabriel39) [inline be/src/vec/sink/vdata_stream_sender.h:511]: Better to use a smart pointer

Comment (HappenLee) [reply be/src/vec/sink/vdata_stream_sender.h:511]: ok

**Comment by HappenLee** [comment]
run buildall

File: be/src/vec/sink/vdata_stream_sender.h [hunk 3]
```diff
@@ -456,6 +420,9 @@ class PipChannel final : public Channel {
         if (_ch_cur_pb_block) {
             delete _ch_cur_pb_block;
         }
+        if (_closure) {
+            delete _closure;
```
Comment (wangbo) [inline be/src/vec/sink/vdata_stream_sender.h:424]: _closure is a unique_ptr，it need not to be delete manually.

**Comment by HappenLee** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**